src/docfx/restore/PackageResolver.cs

```diff
@@ -174,6 +177,67 @@ private void DownloadGitRepositoryCore(string cwd, string url, string committish
             }
         }
 
+        private void ThrowRestoreDependentRepositoryFailureError(string url, string committish, Exception ex)
+        {
+            var systemServiceTokenAccountType = "SystemServiceAccount";
+            var templateRepoUrlPrefix = "https://github.com/Microsoft/templates.docs.msft";
+
+            if (ex.Message.Contains("has enabled or enforced SAML SSO", StringComparison.OrdinalIgnoreCase))
+            {
+                if (url.StartsWith(templateRepoUrlPrefix) || _config.GitAccessTokenAccountType.Equals(systemServiceTokenAccountType))
+                {
+                    throw Errors.System.GitCloneFailed(url, committish, ex).ToException(ex);
+                }
+                else
+                {
+                    throw Errors.CRR.RepositoryOwnerSSOIssue(EnvironmentVariable.RepositoryUrl ?? "this", _config.RepositoryOwnerName, url).ToException(ex);
```

Comment:
what does "this" mean?

Reply:
In case repo url is null, the error message would be "Owner of this repository does not have access to" ->"this" referring to the repo for which build was triggered.
Any suggestion ?

Reply:
In which case would the repo url be null?

Reply:
It's an environment variable, so it shouldn't be null. However, it's defined as nullable type.

Reply:
Updated